src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -42,9 +58,32 @@ internal abstract class AbstractTodoCommentService : ITodoCommentService
         private async Task<IList<TodoComment>> GetTodoCommentsInRemoteHostAsync(
             RemoteHostClient client, Document document, ImmutableArray<TodoCommentDescriptor> commentDescriptors, CancellationToken cancellationToken)
         {
-            return await client.RunCodeAnalysisServiceOnRemoteHostAsync<IList<TodoComment>>(
-                document.Project.Solution, nameof(IRemoteTodoCommentService.GetTodoCommentsAsync),
-                new object[] { document.Id, commentDescriptors }, cancellationToken).ConfigureAwait(false);
+            var keepAliveSession = await TryGetKeepAliveSessionAsync(client).ConfigureAwait(false);
+
+            var (success, result) = await keepAliveSession.TryInvokeAsync<IList<TodoComment>>(
+                nameof(IRemoteTodoCommentService.GetTodoCommentsAsync),
+                document.Project.Solution,
+                new object[] { document.Id, commentDescriptors }).ConfigureAwait(false);
+
+            return success ? result : SpecializedCollections.EmptyList<TodoComment>();
```

Comment:
another example.  instead of this we could get rid of 'success' and just do "result ?? Specialized..."